src/rotator_library/client.py

```
@@ -306,6 +315,71 @@ def __init__(
+            "cooldown_remaining_s": 0.0,
+            "concurrency_slots_in_use": -1,
+            "concurrency_slots_max": self.concurrency_gate._resolve(provider, model)[0],
+        }
+
+    def record_shared_cooldown(
+        self, provider: str, model: str, retry_after_s: float
+    ) -> None:
+        """Persist a 429 cooldown into the cross-process store + release gate."""
+        try:
+            self.shared_cooldown.start_cooldown(provider, model, retry_after_s)
+        except Exception as e:  # pragma: no cover - never let telemetry crash the request path
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)